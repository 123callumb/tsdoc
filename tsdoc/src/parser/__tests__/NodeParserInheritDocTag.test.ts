import { TestHelpers } from './TestHelpers';

test('00 InheritDoc tag: positive examples', () => {
  TestHelpers.parseAndMatchNodeParserSnapshot([
    '/**',
    ' * {@inheritDoc}',
    ' * {@inheritDoc Class.member}',
    ' * {@inheritDoc package# Class . member}',
    ' */'
  ].join('\n'));
});

test('01 InheritDoc tag: negative examples', () => {
  TestHelpers.parseAndMatchNodeParserSnapshot([
    '/**',
    ' * {@inheritDoc | link text}',
    ' */'
  ].join('\n'));
});
