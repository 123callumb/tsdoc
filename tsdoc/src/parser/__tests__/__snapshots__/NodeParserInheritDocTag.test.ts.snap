// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`00 InheritDoc tag: positive examples 1`] = `
Object {
  "buffer": "/**[n] * {@inheritDoc}[n] */",
  "gaps": Array [],
  "lines": Array [
    "{@inheritDoc}",
  ],
  "logMessages": Array [],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "kind": "InheritDocTag",
        "nodes": Array [
          Object {
            "kind": "Excerpt: InlineTag_OpeningDelimiter",
            "nodeExcerpt": "{",
          },
          Object {
            "kind": "Excerpt: InlineTag_TagName",
            "nodeExcerpt": "@inheritDoc",
          },
          Object {
            "kind": "Excerpt: InlineTag_ClosingDelimiter",
            "nodeExcerpt": "}",
          },
        ],
      },
    ],
  },
}
`;

exports[`00 InheritDoc tag: positive examples 2`] = `
Object {
  "buffer": "/**[n] * {@inheritDoc Class.member}[n] */",
  "gaps": Array [],
  "lines": Array [
    "{@inheritDoc Class.member}",
  ],
  "logMessages": Array [],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "kind": "InheritDocTag",
        "nodes": Array [
          Object {
            "kind": "Excerpt: InlineTag_OpeningDelimiter",
            "nodeExcerpt": "{",
          },
          Object {
            "kind": "Excerpt: InlineTag_TagName",
            "nodeExcerpt": "@inheritDoc",
          },
          Object {
            "kind": "Excerpt: Spacing",
            "nodeExcerpt": " ",
          },
          Object {
            "kind": "DeclarationReference",
            "nodes": Array [
              Object {
                "kind": "MemberReference",
                "nodes": Array [
                  Object {
                    "kind": "MemberIdentifier",
                    "nodes": Array [
                      Object {
                        "kind": "Excerpt: MemberIdentifier_Identifier",
                        "nodeExcerpt": "Class",
                      },
                    ],
                  },
                ],
              },
              Object {
                "kind": "MemberReference",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: MemberReference_Dot",
                    "nodeExcerpt": ".",
                  },
                  Object {
                    "kind": "MemberIdentifier",
                    "nodes": Array [
                      Object {
                        "kind": "Excerpt: MemberIdentifier_Identifier",
                        "nodeExcerpt": "member",
                      },
                    ],
                  },
                ],
              },
            ],
          },
          Object {
            "kind": "Excerpt: InlineTag_ClosingDelimiter",
            "nodeExcerpt": "}",
          },
        ],
      },
    ],
  },
}
`;

exports[`00 InheritDoc tag: positive examples 3`] = `
Object {
  "buffer": "/**[n] * {@inheritDoc package# Class . member}[n] */",
  "gaps": Array [],
  "lines": Array [
    "{@inheritDoc package# Class . member}",
  ],
  "logMessages": Array [],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "kind": "InheritDocTag",
        "nodes": Array [
          Object {
            "kind": "Excerpt: InlineTag_OpeningDelimiter",
            "nodeExcerpt": "{",
          },
          Object {
            "kind": "Excerpt: InlineTag_TagName",
            "nodeExcerpt": "@inheritDoc",
          },
          Object {
            "kind": "Excerpt: Spacing",
            "nodeExcerpt": " ",
          },
          Object {
            "kind": "DeclarationReference",
            "nodes": Array [
              Object {
                "kind": "Excerpt: DeclarationReference_PackageName",
                "nodeExcerpt": "package",
              },
              Object {
                "kind": "Excerpt: DeclarationReference_ImportHash",
                "nodeExcerpt": "#",
              },
              Object {
                "kind": "Excerpt: Spacing",
                "nodeExcerpt": " ",
              },
              Object {
                "kind": "MemberReference",
                "nodes": Array [
                  Object {
                    "kind": "MemberIdentifier",
                    "nodes": Array [
                      Object {
                        "kind": "Excerpt: MemberIdentifier_Identifier",
                        "nodeExcerpt": "Class",
                      },
                    ],
                  },
                  Object {
                    "kind": "Excerpt: Spacing",
                    "nodeExcerpt": " ",
                  },
                ],
              },
              Object {
                "kind": "MemberReference",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: MemberReference_Dot",
                    "nodeExcerpt": ".",
                  },
                  Object {
                    "kind": "Excerpt: Spacing",
                    "nodeExcerpt": " ",
                  },
                  Object {
                    "kind": "MemberIdentifier",
                    "nodes": Array [
                      Object {
                        "kind": "Excerpt: MemberIdentifier_Identifier",
                        "nodeExcerpt": "member",
                      },
                    ],
                  },
                ],
              },
            ],
          },
          Object {
            "kind": "Excerpt: InlineTag_ClosingDelimiter",
            "nodeExcerpt": "}",
          },
        ],
      },
    ],
  },
}
`;

exports[`01 InheritDoc tag: negative examples 1`] = `
Object {
  "buffer": "/**[n] * {@inheritDoc | link text}[n] */",
  "gaps": Array [],
  "lines": Array [
    "{@inheritDoc | link text}",
  ],
  "logMessages": Array [
    "(2,5): Expecting a declaration reference",
  ],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "InlineTag",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: InlineTag_OpeningDelimiter",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Excerpt: InlineTag_TagName",
                    "nodeExcerpt": "@inheritDoc",
                  },
                  Object {
                    "kind": "Excerpt: Spacing",
                    "nodeExcerpt": " ",
                  },
                  Object {
                    "kind": "Excerpt: InlineTag_TagContent",
                    "nodeExcerpt": "| link text",
                  },
                  Object {
                    "kind": "Excerpt: InlineTag_ClosingDelimiter",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
}
`;

exports[`01 InheritDoc tag: negative examples 2`] = `
Object {
  "buffer": "/**[n] * {@inheritDoc Class % junk}[n] */",
  "gaps": Array [],
  "lines": Array [
    "{@inheritDoc Class % junk}",
  ],
  "logMessages": Array [
    "(2,23): Unexpected character after declaration reference",
  ],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "InlineTag",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: InlineTag_OpeningDelimiter",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Excerpt: InlineTag_TagName",
                    "nodeExcerpt": "@inheritDoc",
                  },
                  Object {
                    "kind": "Excerpt: Spacing",
                    "nodeExcerpt": " ",
                  },
                  Object {
                    "kind": "Excerpt: InlineTag_TagContent",
                    "nodeExcerpt": "Class % junk",
                  },
                  Object {
                    "kind": "Excerpt: InlineTag_ClosingDelimiter",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
}
`;

exports[`01 InheritDoc tag: negative examples 3`] = `
Object {
  "buffer": "/**[n] * {@inheritDoc}[n] * {@inheritDoc}[n] */",
  "gaps": Array [],
  "lines": Array [
    "{@inheritDoc}",
    "{@inheritDoc}",
  ],
  "logMessages": Array [
    "(3,4): A doc comment cannot have more than one @inheritDoc tag",
  ],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
              Object {
                "errorLocation": "{@inheritDoc}",
                "errorLocationPrecedingToken": "
",
                "errorMessage": "A doc comment cannot have more than one @inheritDoc tag",
                "kind": "ErrorText",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: ErrorText",
                    "nodeExcerpt": "{@inheritDoc}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "kind": "InheritDocTag",
        "nodes": Array [
          Object {
            "kind": "Excerpt: InlineTag_OpeningDelimiter",
            "nodeExcerpt": "{",
          },
          Object {
            "kind": "Excerpt: InlineTag_TagName",
            "nodeExcerpt": "@inheritDoc",
          },
          Object {
            "kind": "Excerpt: InlineTag_ClosingDelimiter",
            "nodeExcerpt": "}",
          },
        ],
      },
    ],
  },
}
`;

exports[`01 InheritDoc tag: negative examples 4`] = `
Object {
  "buffer": "/**[n] * summary text[n] * @remarks[n] * {@inheritDoc}[n] */",
  "gaps": Array [],
  "lines": Array [
    "summary text",
    "@remarks",
    "{@inheritDoc}",
  ],
  "logMessages": Array [
    "(3,4): A \\"@remarks\\" block must not be used, because that content is provided by the @inheritDoc tag",
    "(1,1): The summary section must not have any content, because that content is provided by the @inheritDoc tag",
  ],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "PlainText",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: PlainText",
                    "nodeExcerpt": "summary text",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "kind": "Block",
        "nodes": Array [
          Object {
            "kind": "BlockTag",
            "nodes": Array [
              Object {
                "kind": "Excerpt: BlockTag",
                "nodeExcerpt": "@remarks",
              },
            ],
          },
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "kind": "InheritDocTag",
        "nodes": Array [
          Object {
            "kind": "Excerpt: InlineTag_OpeningDelimiter",
            "nodeExcerpt": "{",
          },
          Object {
            "kind": "Excerpt: InlineTag_TagName",
            "nodeExcerpt": "@inheritDoc",
          },
          Object {
            "kind": "Excerpt: InlineTag_ClosingDelimiter",
            "nodeExcerpt": "}",
          },
        ],
      },
    ],
  },
}
`;

exports[`01 InheritDoc tag: negative examples 5`] = `
Object {
  "buffer": "/**[n] * {@inheritdoc @scope/library:IDisposable.isDisposed}[n] */",
  "gaps": Array [],
  "lines": Array [
    "{@inheritdoc @scope/library:IDisposable.isDisposed}",
  ],
  "logMessages": Array [
    "(2,17): The declaration reference appears to contain a package name or import path, but it is missing the \\"#\\" delimiter",
  ],
  "nodes": Object {
    "kind": "Comment",
    "nodes": Array [
      Object {
        "kind": "Section",
        "nodes": Array [
          Object {
            "kind": "Paragraph",
            "nodes": Array [
              Object {
                "kind": "InlineTag",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: InlineTag_OpeningDelimiter",
                    "nodeExcerpt": "{",
                  },
                  Object {
                    "kind": "Excerpt: InlineTag_TagName",
                    "nodeExcerpt": "@inheritdoc",
                  },
                  Object {
                    "kind": "Excerpt: Spacing",
                    "nodeExcerpt": " ",
                  },
                  Object {
                    "kind": "Excerpt: InlineTag_TagContent",
                    "nodeExcerpt": "@scope/library:IDisposable.isDisposed",
                  },
                  Object {
                    "kind": "Excerpt: InlineTag_ClosingDelimiter",
                    "nodeExcerpt": "}",
                  },
                ],
              },
              Object {
                "kind": "SoftBreak",
                "nodes": Array [
                  Object {
                    "kind": "Excerpt: SoftBreak",
                    "nodeExcerpt": "[n]",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
}
`;
